README.md

```diff
@@ -0,0 +1,180 @@
+# @apostrophecms/content-upgrader
+
+A tool to migrate your content from Apostrophe 2.x to Apostrophe 3.x.
+
+This tool installs as a module **inside your existing A2 project.** It will export content to a separate database for use in a new A3 project.
+
+## Installation
+
+```bash
+cd my-existing-a2-project
+npm install @apostrophecms/content-upgrader
+```
+
+## Configuration
+
+Edit `app.js` of your A2 project. Add the module to the `modules` section:
+
+```javascript
+modules: {
+  // ... all of your existing A2 modules here ...
+  '@apostrophecms/content-upgrader': {}
+}
+```
+
+Then create `lib/modules/@apostrophecms/content-upgrader/index.js`. Here you can optionally address any changes in doc type and widget type names and do deeper transformations as well.
+
+### Transforming doc types
+
+In A2, the names of piece type modules themselves and the `name` option configured for the module were often different. By convention the piece type module name was plural and the `name` option was singular. The `name` option is what is used in the database to set the `type` property of each piece.
+
+Yes, this was confusing. That's why in A3, the `name` option no longer exists, and **the name of the module and the `type` property in the database are always the same.** Also, by convention, module names in A3 are singular.
+
+What does this mean for us when we upgrade the content? It means that most of the time, **we need to change the module name to singular and remove its `name` option for A3, but we don't need to change the `type` in the database.**
+
+However, if you *do* want to remap the `type` in the database, perhaps because you're choosing a different name for your module in A3, you can do it like this:
+
+```javascript
+module.exports = {
+  mapDocTypes: {
+    'old-name': 'new-name',
+    ...
+  }
+}
+```
+
+> `old-name` must match the `name` option from the old A2 module, *not* the module name from A2. `new-name` must match the module name in your new A3 project, which will also be used for the `type` property.
```

Comment:
Stylistically I think we should stop using blockquotes for things like this. Block quotes are simply a different thing than a callout. Bolding, a "Note:" preamble, or some other technique is better.

Reply:
There are a few more below.